File: AOJ/0346/0346.c
#include<stdio.h>
#define N 12
int e[N];

int check(){
  int i, j, minj, t;
  /* 選択ソート */
  for ( i = 0; i < N - 1; i++ ){
    minj = i;
    for ( j = i + 1; j < N; j++ )
      if ( e[j] < e[minj] ) minj = j;
    t = e[minj];
    e[minj] = e[i];
    e[i] = t;
  }
  
  for ( i = 0; i < N; i += 4)
    if ( e[i] != e[i + 3] ) return 0;
  return 1;
}

int main(){
  int i;
  for ( i = 0; i < N; i++ ) scanf("%d", &e[i]);
  printf(check() ? "yes\n": "no\n");
  return 0;
}

File: AOJ/0346/0346.py
N = 12
e = list(map(int, input().split()))
# 選択ソート
for i in range(N -1):
    minj = i
    for j in range(i + 1, N):
        if  e[j] < e[minj]: minj = j
    e[minj], e[i] = e[i], e[minj]
print("yes" if e[0] == e[3] and e[4] == e[7] and e[8] == e[11] else "no")



